### PR TITLE
chore: add publishers and subscriptions properties

### PR DESCRIPTION
**Pull request by hsgwa**
This PR adds `publishers` and `subscriptions` properties.

- app.publishers / app.subscriptions
- arch.publishers / arch.subscriptions

After this PR is merged, #180 can be merged.

### CODE REVIEW

File: src/caret_analyze/architecture/architecture.py [hunk 1]
```diff
@@ -139,6 +140,16 @@ def path_names(self) -> Tuple[str, ...]:
     def communications(self) -> Tuple[CommunicationStructValue, ...]:
         return tuple([v.to_value() for v in self._communications])
 
+    @property
+    def publishers(self) -> Tuple[PublisherStructValue, ...]:
```
Comment (keita1523) [inline src/caret_analyze/architecture/architecture.py:144]: I have a question.
`Architecture.publishers/subscriptions` returns as Tuple and `Application.publishers/subscriptions` returns as List.
Why is there difference?

Comment (hsgwa) [reply src/caret_analyze/architecture/architecture.py:144]: Architecture class returns **immutable** instances (tuple).
Application class returns **mutables** instances (list).

https://tier4.github.io/CARET_doc/latest/design/software_architecture/caret_analyze/#architecture
> All information retrieved from the Architecture file is of type ValueObject, which is suitable for interfacing information between other packages.

**Review by keita1523 — APPROVED** [review]
LGTM.

**Review by takeshi-iwanari — APPROVED** [review]
LGTM

**Pull request merged**